spec/avram/operations/save_operation_spec.cr

```diff
@@ -307,6 +312,126 @@ describe "Avram::SaveOperation" do
     end
   end
 
+  describe ".upsert" do
+    # TODO: Is this possible? How do we write this?
+    it "should only proc one query" do
+    end
+
+    context "when a record already exists" do
+      before_each do
+        UserFactory.create do |u|
+          u.name("Name 1")
+          u.nickname("Nickname 1")
+          u.age(42)
+          u.year_born(1960)
+          u.joined_at(Time.utc)
+        end
+      end
+
+      it "allows manual passing of updated_at, but ignores created_at" do
+        some_time = Time.utc(2016, 2, 15, 10, 20, 30)
```

Comment:
Fun fact, I took this from the crystal docs for Time, and so did a lot of other people already: https://cs.github.com/?scopeName=All+repos&scope=&q=Time.utc%282016%2C+2%2C+15%2C+10%2C+20%2C+30%29